### PR TITLE
feat(helm): add extra annotation on deployment

### PR DESCRIPTION
**Pull request by sebglon**
Add extra annotation from helm values on deployment.

Closes https://github.com/hashicorp/vault-secrets-operator/issues/168

### CODE REVIEW

**Comment by kschoche** [comment]
Hi @sebglon -
Thanks for filing this PR!
Would you be able to add some bats testing alongside the template changes?
You'd need to add it to `test/unit/deployment.bats` and probably two tests: one for default behaviour of no annotations being set, and one for them being set.
Thanks!

**Comment by sebglon** [comment]
@kschoche thanks for your feedback.
tests added.

File: test/unit/deployment.bats [hunk 1]
```diff
@@ -156,4 +156,30 @@ load _helpers
...
+
+@test "controller/Deployment: annotations not set by default" {
+  cd `chart_dir`
+  local object=$(helm template \
+      -s templates/deployment.yaml  \
+      . | tee /dev/stderr |
+      yq '.spec.template.metadata.annotations | select(documentIndex == 1)' | tee /dev/stderr)#
+
+   local actual=$(echo "$object" | yq '. | length' | tee /dev/stderr)
+    [ "${actual}" = "1" ]
```
[the remark this answers is not in the document]
Comment (sebglon) [reply test/unit/deployment.bats:170]: done

File: test/unit/deployment.bats [hunk 2]
```diff
@@ -156,4 +156,30 @@ load _helpers
...
+  local object=$(helm template \
+      -s templates/deployment.yaml  \
+      --set 'controller.annotations.annot1=value1' \
+      . | tee /dev/stderr |
+      yq '.spec.template.metadata.annotations | select(documentIndex == 1)' | tee /dev/stderr)
+
+   local actual=$(echo "$object" | yq '. | length' | tee /dev/stderr)
+    [ "${actual}" = "2" ]
+   actual=$(echo "$object" | yq '.annot1' | tee /dev/stderr)
+    [ "${actual}" = "value1" ]
```
[the remark this answers is not in the document]
Comment (sebglon) [reply test/unit/deployment.bats:189]: done

File: test/unit/deployment.bats [hunk 3]
```diff
@@ -156,4 +156,30 @@ load _helpers
...
+
+   local actual=$(echo "$object" | yq '. | length' | tee /dev/stderr)
+    [ "${actual}" = "1" ]
+}
 
+@test "controller/Deployment: annotations can be set" {
+  cd `chart_dir`
+  local object=$(helm template \
+      -s templates/deployment.yaml  \
+      --set 'controller.annotations.annot1=value1' \
```
[the remark this answers is not in the document]
Comment (sebglon) [reply test/unit/deployment.bats:179]: done

File: chart/values.yaml [hunk 4]
```diff
@@ -30,8 +30,14 @@ controller:
         cpu: 5m
         memory: 64Mi
 
+  # This value defines additional annotations for the deployment. This should be formatted as a multi-line
+  # string.
+  annotations: null
```
[the remark this answers is not in the document]
Comment (sebglon) [reply chart/values.yaml]: done

File: chart/values.yaml [hunk 5]
```diff
@@ -30,8 +30,14 @@ controller:
...
+
   # Settings related to the vault-secrets-operator container.
   manager:
+
+
```
[the remark this answers is not in the document]
Comment (sebglon) [reply chart/values.yaml]: done

File: chart/values.yaml [hunk 6]
```diff
@@ -30,8 +30,14 @@ controller:
         cpu: 5m
         memory: 64Mi
 
+  # This value defines additional annotations for the deployment. This should be formatted as a multi-line
+  # string.
```
[the remark this answers is not in the document]
Comment (sebglon) [reply chart/values.yaml]: done

**Comment by robertfairhead** [comment]
@benashz Any updates on this? This is a blocker for us to move forward with a broader roll out of the operator.

**Comment by benashz** [comment]
Thanks for making the requested changes @sebglon . If you wouldn't mind resolving the merge conflicts, I'd be happy to get this merged for you.

**Comment by sebglon** [comment]
@benashz  conflict merge solved

**Comment by benashz** [comment]
> @benashz Any updates on this? This is a blocker for us to move forward with a broader roll out of the operator.

Hey @robertfairhead , thanks for the interest in the Operator. We should be able to get this merged once the conflicts are resolved.

On another note, there will be some upcoming breaking changes to the CRD specs, e.g: #171 . Since we are still in beta mode we will not be providing multi CRD version support. Post GA, we will be providing seamless CRD version upgrades and backwards compatibility.

**Review by benashz — APPROVED** [review]
Looks good! Thank you for your contribution to HashiCorp!

**Comment by robertfairhead** [comment]
Yeah, understood re: CRDs changing. We have it on one test cluster but want to expand to a few more so some additional internal teams can test. We're not going to prod until it's GA but want to get it into internal beta.

**Pull request merged**